Added docs and js example for REST API usage (#104)
jenkinsci/file-parameters-plugin: README.md

build job: 'downstream', parameters: [base64File(name: 'file', base64: Base64.encoder.encodeToString('hello'.bytes)))]
```

## Usage with REST API

> You can pass file parameters to the REST API:

Reviewer: `buildWithParameters`, especially with file uploads, seems to strain the definition of https://en.wikipedia.org/wiki/Representational_state_transfer#Architectural_constraints. https://www.jenkins.io/doc/book/using/remote-access-api/ does refer to a _REST-like style_ though actually this refers to the read-only “export” system https://javadoc.jenkins.io/component/stapler/org/kohsuke/stapler/export/package-summary.html + https://javadoc.jenkins.io/hudson/model/Api.html. Nitpicking but maybe the wording can be relaxed to just say a “remote” or “HTTP” API?

Author: I can certainly do that. Although since Jenkins itself seems to refer to `buildWithParameters` as part of their REST API, it will probably be a good idea to at least have that name in a parenthetical note to help people to land in a successful conclusion

```js
const file = fileInput.files[0]; // a File object
const body = new FormData();
body.append('ICON_FILE', file); // will come through to the job as the named file parameter 'ICON_FILE'
const request = new Request(`${buildAPI}buildWithParameters`, { method: 'POST', body });

> fetch(request); // omitted, crumb header and other credentials

Reviewer: You should not need a crumb header—use an API token.

Author: Agreed

Author: > Good to document, though the example presumes familiarity with what I presume is Node.js. 

Actually it's really just a working example from js running in browser. I think something similar would also work in recent versions of nodejs as `fetch` has only been recently added. I'll add the `curl` example as well, but I don't really have a good way to test it.

Reviewer: Got it, thanks.

build job: 'downstream', parameters: [base64File(name: 'file', base64: Base64.encoder.encodeToString('hello'.bytes)))]
```

## Usage with HTTP API

> You can pass file parameters to the HTTP API (in the Jenkins UI, this HTTP API is also referred to as "REST API"):

Reviewer: ```suggestion
You can pass file parameters to the HTTP API (in the Jenkins UI, this HTTP API is also referred to as “REST API”):
```

Curl example:

> ```

Reviewer: ```suggestion
```bash
```

curl -u $auth -F FILE=@/tmp/f $jenkins/job/myjob/buildWithParameters
```

Javascript example:

```js
const file = fileInput.files[0]; // a File object
const body = new FormData();

> body.append('ICON_FILE', file); // will come through to the job as the named file parameter 'ICON_FILE'

Reviewer: ```suggestion
body.append('FILE', file); // will come through to the job as the named file parameter 'FILE'
```

(just to be more generic, and to parallel the `curl` example)

> const request = new Request(`${buildAPI}buildWithParameters`, { method: 'POST', body });

Reviewer: ```suggestion
const request = new Request(`${jobUrl}buildWithParameters`, { method: 'POST', body });
```

would be more comprehensible I think? Where `jobUrl` is something like `https://jenkins/job/myjob/`
